### PR TITLE
Fix/tx fee checks overview

### PR DESCRIPTION
**Pull request by sahilnanda1995**

### CODE REVIEW

File: components/overview/InvestMoreModal.js [hunk 1]
```diff
@@ -368,16 +379,29 @@ const InvestMoreModal = withSlideIn(
...
+													</div>
 												) : (
-													amount >
-														balance?.availableBalance /
-															Math.pow(10, networkInfo.decimalPlaces) -
-															networkInfo.reserveAmount && (
+													controllerAccount &&
+													controllerBalances &&
+													transactionFee +
+														apiInstance?.consts.balances.existentialDeposit.toNumber() >
```
Comment (sahilnanda1995) [inline components/overview/InvestMoreModal.js:397]: change this to `transactionFee + (apiInstance?.consts.balances.existentialDeposit.toNumber() / 2)`

File: components/overview/ReBondModal.js [hunk 2]
```diff
@@ -157,6 +160,14 @@ const ReBondModal = withSlideIn(
 					minPossibleStake
 			) {
 				setCalculationDisabled(true);
+			} else if (
+				controllerAccount &&
+				controllerBalances &&
+				transactionFee +
+					apiInstance?.consts.balances.existentialDeposit.toNumber() >
```
Comment (sahilnanda1995) [inline components/overview/ReBondModal.js:167]: same as in InvestMore, change this to `transactionFee + (apiInstance?.consts.balances.existentialDeposit.toNumber() / 2)`

File: components/overview/RedeemUnbonded.js [hunk 3]
```diff
@@ -198,6 +202,16 @@ const RedeemUnbonded = withSlideIn(
 								<>
 									<h3 className="mt-4 text-2xl">Confirmation</h3>
 									<div className="w-full mt-8">
+										{controllerAccount &&
+											controllerBalances &&
+											transactionFee +
+												apiInstance?.consts.balances.existentialDeposit.toNumber() >
```
Comment (sahilnanda1995) [inline components/overview/RedeemUnbonded.js:208]: change this to transactionFee + (apiInstance?.consts.balances.existentialDeposit.toNumber() / 2)

File: components/overview/RedeemUnbonded.js [hunk 4]
```diff
@@ -236,8 +250,23 @@ const RedeemUnbonded = withSlideIn(
 									</div>
 									<div className="w-full flex-center">
 										<button
-											className="rounded-full font-medium px-12 py-3 bg-teal-500 mt-40 mb-40 text-white"
+											className={`rounded-full font-medium px-12 py-3 ${
+												controllerAccount &&
+												controllerBalances &&
+												transactionFee +
+													apiInstance?.consts.balances.existentialDeposit.toNumber() >
```
Comment (sahilnanda1995) [inline components/overview/RedeemUnbonded.js:257]: change this to transactionFee + (apiInstance?.consts.balances.existentialDeposit.toNumber() / 2)

File: components/overview/WithdrawModal.js [hunk 5]
```diff
@@ -588,6 +591,14 @@ const WithdrawModal = withSlideIn(
 					Math.pow(10, networkInfo.decimalPlaces)
 			) {
 				setCalculationDisabled(true);
+			} else if (
+				controllerAccount &&
+				controllerBalances &&
+				totalTransactionFee +
+					apiInstance?.consts.balances.existentialDeposit.toNumber() >
```
Comment (sahilnanda1995) [inline components/overview/WithdrawModal.js:598]: change this to transactionFee + (apiInstance?.consts.balances.existentialDeposit.toNumber() / 2)

File: components/overview/InvestMoreModal.js [hunk 6]
```diff
@@ -162,6 +165,14 @@ const InvestMoreModal = withSlideIn(
 					minPossibleStake
 			) {
 				setCalculationDisabled(true);
+			} else if (
+				controllerAccount &&
```
[the remark this answers is not in the document]
Comment (Lord-of-Codes) [reply components/overview/InvestMoreModal.js:169]: oh yes. missed this. the dispatch origin must be stash.

File: components/overview/InvestMoreModal.js (hunk 1 above)
Comment (Lord-of-Codes) [reply components/overview/InvestMoreModal.js:397]: @sahilnanda1995 i do not understand this. if a user is using a separate controller then that account will get reaped if its balance drops below the existential deposit.

Comment (sahilnanda1995) [reply components/overview/InvestMoreModal.js:397]: > then that account will get reaped if its balance drops below the existential deposit
hmm right, i was somehow thinking of including a total balance check for an account but that will be low yield

Can you also crosscheck this and push a fix for the same for staking, reward-calculator and validators page module

Comment (Lord-of-Codes) [reply components/overview/InvestMoreModal.js:397]: @sahilnanda1995 I think its already being checked in validator and reward-calculator page but I will check them.

File: components/overview/InvestMoreModal.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (Lord-of-Codes) [reply components/overview/InvestMoreModal.js:169]: @sahilnanda1995 i will remove the controller check here. stash is already being checked for the reserve amount.

[the remark this answers is not in the document]
Comment (sahilnanda1995) [reply components/overview/InvestMoreModal.js:169]: yep, no connection to controller account for bondExtra

**Pull request merged**